File: fx/src/main/java/gargoyle/fx/FXImageSplash.java
package gargoyle.fx;

import gargoyle.fx.log.FXLog;

import javax.swing.ImageIcon;
import javax.swing.JLabel;
import javax.swing.JWindow;
import java.awt.BorderLayout;
import java.awt.Color;
import java.awt.Container;
import java.awt.Window;
import java.net.URL;
import java.util.Map;
import java.util.ResourceBundle;

public final class FXImageSplash implements FXSplash {
    private static final Color TRANSPARENT = new Color(0, 0, 0, 0);

    public FXImageSplash() {
    }

    @Override
    public Window createWindow(final URL location, final ResourceBundle resources) {
        final JWindow window = new JWindow();
        final Container pane = window.getContentPane();
        pane.setLayout(new BorderLayout());
        final JLabel label = new JLabel(new ImageIcon(location));
        label.setBackground(TRANSPARENT);
        pane.add(label, BorderLayout.CENTER);
        window.setBackground(TRANSPARENT);
        return window;
    }

    @Override
    public void handleSplashNotification(final FXSplashNotification splashNotification) {
        FXLog.info("{name}.{type}:{progress}({details})",
                Map.of("name", splashNotification.getClass().getSimpleName(),
                        "type", splashNotification.getType(),
                        "progress", splashNotification.getProgress(),
                        "details", splashNotification.getDetails())
        );
    }
}

File: fx/src/main/kotlin/gargoyle/fx/FxImageSplash.kt
package gargoyle.fx

import gargoyle.fx.FxSplash.FxSplashNotification
import gargoyle.fx.log.FxLog
import java.awt.BorderLayout
import java.awt.Color
import java.awt.Window
import java.net.URL
import java.util.ResourceBundle
import javax.swing.ImageIcon
import javax.swing.JLabel
import javax.swing.JWindow

class FxImageSplash : FxSplash {
    override fun createWindow(location: URL, resources: ResourceBundle?): Window =
        JWindow().also { window ->
            with(window.contentPane.also { it.layout = BorderLayout() }) {
                add(JLabel(ImageIcon(location)).also { background = TRANSPARENT }, BorderLayout.CENTER)
            }
        }.also { it.background = TRANSPARENT }

    override fun handleSplashNotification(splashNotification: FxSplashNotification): Unit =
        FxLog.info("${splashNotification.javaClass.simpleName}.${splashNotification.type}:${splashNotification.progress}(${splashNotification.details})")

    companion object {
        private val TRANSPARENT = Color(0, 0, 0, 0)
    }
}
